core/src/com/unciv/logic/automation/ConstructionAutomation.kt

```diff
@@ -322,4 +338,47 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
         }
     }
 
+    private fun addReligousUnit(){
+
+        var modifier = 0f
+
+        val missionary = cityInfo.getRuleset().units.values.asSequence()
+            .filter { it.canBePurchasedWithStat(cityInfo, Stat.Faith) }
+            .filter { it.hasUnique("can spread religion")}
+            .firstOrNull()
+
+        val inquisitor = cityInfo.getRuleset().units.values.asSequence()
+            .filter { it.canBePurchasedWithStat(cityInfo, Stat.Faith) }
+            .filter { it.hasUnique("can remove hearsay") }
+            .firstOrNull()
```

Comment:
Granted I've not touched the Uniques system much at all, but I'm not seeing anything matching either of these two Unique texts in the Units.json:

https://github.com/yairm210/Unciv/blob/b72a2c3ebbed5f12d47768000db82b64e61c7b4e/android/assets/jsons/Civ%20V%20-%20Gods%20%26%20Kings/Units.json#L1654-L1672

If you want to compare by string, then it looks like the `.placeHolderText` is "Can [] [] times":

![image](https://user-images.githubusercontent.com/37680486/145283316-813bccf4-d159-4a6e-a74e-64c548565203.png)

However, because the action in question is given as a parameter, it looks like all Missionaries, Inquisitors, and future units with limited actions all use the same unique, so you'd also have to check the `Unique.params`:

![image](https://user-images.githubusercontent.com/37680486/145284222-81077760-9cb1-4793-b090-f3e8c4774920.png)

Hm. Someone that actually knows the code can correct me, but I think it should work to do something like `.getMatchingUniques("Can [] [] times").any{ it.params[0] == "Spread Religion" }`? Or maybe just `.uniqueObjects.any{}` to roll the two checks together, I think.

https://github.com/yairm210/Unciv/blob/master/core/src/com/unciv/models/ruleset/IHasUniques.kt

---

Also, can't you do a single `.firstOrNull { it.canBePurchased… && it.hasUnique… }`, instead of the `asSequence().filter{}.filter{}.firstOrNull`?

Reply:
```
.getMatchingUniques("Can [] [] times").any { it.params[0] == "Spread Religion" }
```
is indeed the 'usual' way uniques are checked

Reply:
Fixed

Reply:
I do see any new commits yet?